apps/teamsjs-cc-app/tabs/src/components/capabilities/AppInstallDialog.tsx

```diff
@@ -0,0 +1,35 @@
+import { app, appInstallDialog } from "@microsoft/teams-js";
+
+import { Button } from "@fluentui/react-northstar";
+import { booleanToString } from "../../helpers";
+import { developersPortalAppId } from "../../helpers/constants";
+
+/**
+ * This component Open's a dialog with particular application to install.
+ */
+export const AppInstallDialog = () => {
+  // Check if app is initialized
+  if (app.isInitialized()) {
+    // check to see if capability is supported
+    if (appInstallDialog.isSupported()) {
+      // return button to open dialog
+      return (
+        <Button
+          onClick={async () => {
+            // open the install dialog for the Developer Portal app
+            await appInstallDialog.openAppInstallDialog({
+              appId: developersPortalAppId,
+            });
+          }}
+        >
+          Open App Install Dialog
+        </Button>
+      );
+    }
+  }
+  // return's if capability is not supported.
+  return <>Capability is not supported</>;
```

Comment:
I think this is actually "Capability is not suported OR app is not initialized". Might be good to return different values in those cases to help people understand why it doesn't work (this comment probably applies to a lot of the files in here.

Also, if that string appears in a lot of files may be worth putting it in your constants file so that future changes to it are easier.

Reply:
It looks like you've actually done this correctly in some (most?) of the other files, so maybe just a few missed ones like this/BarCode